magento-research/pwa-studio
# [feature]: Adding a window size hook

## Description
I added a window size hook so that components that need to do something based on window size (recalc/rerender/etc) can do so without having to reimplement the same functionality over and over.

I refactored the `Thumbnail` component to use the hook in determining whether or not to render the image.

## Related Issue





Closes #1192 .

## Verification Steps
1) Open the app using chrome dev tools and set your display to "iPad" in portrait mode.
2) Navigate to a PDP and see that no thumbnails are requested.
3) Turn the ipad to landscape mode
4) See that thumbnails are requested (240px image).

## How Have YOU Tested this?
Above verification steps.

## Screenshots / Screen Captures (if appropriate):

## Proposed Labels for Change Type/Package



- [x] major (e.g x.0.0 - a breaking change) * ONLY MAJOR BECAUSE `useDocumentListener` -> `useEventListener`

## Checklist:
